src/cmd/root.go

```diff
@@ -126,6 +116,31 @@ func run(cmd *cobra.Command, _ []string) {
 	}
 }
 
+// NewZarfCommand creates the `zarf` command and its nested children.
+func NewZarfCommand() *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:          "zarf COMMAND",
+		Short:        lang.RootCmdShort,
+		Long:         lang.RootCmdLong,
+		Args:         cobra.MaximumNArgs(1),
+		SilenceUsage: true,
+		// TODO(mkcp): Do we actually want to silence errors here?
+		SilenceErrors:     true,
+		PersistentPreRunE: preRun,
+		Run:               run,
+	}
+
+	// TODO(soltysh): consider adding command groups
+	rootCmd.AddCommand(NewConnectCommand())
+	rootCmd.AddCommand(NewDestroyCommand())
+	rootCmd.AddCommand(NewDevCommand())
+	rootCmd.AddCommand(NewInitCommand())
+	rootCmd.AddCommand(NewInternalCommand(rootCmd))
+	rootCmd.AddCommand(NewPackageCommand())
```

Comment:
This is the biggest win of the refactor imo -- we don't need to understand what magic implicit invocation order you get from a bunch of init()s being executed. Clear and declarative 👍 

One note though, `zarf say`'s command should probably be added to this sequence. It's currently in Execute below, but it's better off grouped with everything else.

Reply:
Yes!

The other part I actually started considering is grouping commands, have a look at `kubectl` output, it groups commands into several groups (basic, deploy, cluster management, etc...). We don't have that many, yet, but if we want to consider this construct will make it very easy to do just that.

Reply:
That makes a lot of sense, i'm in favor!